modules/gmake/gmake.lua

```diff
@@ -0,0 +1,333 @@
+---
+-- GNU Makefile exporter.
+---
+
+local dom = require('dom')
+local path = require('path')
+local premake = require('premake')
+local State = require('state')
+
+local gmake = {}
+
+gmake.sln = doFile('./src/sln.lua', gmake)
+gmake.proj = doFile('./src/proj.lua', gmake)
+
+local _ARCHITECTURES = {
```

Comment:
This shouldn't be `local`, doing this makes it impossible to extend from a module or script. Also, this isn't used anywhere.

Reply:
Tbh, this was just copied from the vstudio module, but you're right, it's unused. Will remove.

Reply:
> **EDIT**: I did find a usage. We use this to look up the architecture instead of assuming that the value in the premake api is the value that is passed directly to the makefile.

VS requires specific values for the architectures, we definitely shouldn't be bringing that concept to any generator that doesn't need it. I don't want to update Premake just to get access to a new architecture, I want to just add a value into an array and be done with it. Also, you can't write modules to add support for other systems if this is how things work.

Additionally, for VS, we should be doing `lookupTable[cfg.architecture] or cfg.architecture`. The "win32 hack" in Premake5 was a pain for me every single time I tried to extend the vstudio module to support a new platform/architecture. I understand _why_ the VS module has the table, but it shouldn't be `local` and we shouldn't be defaulting to `Win32` we should be defaulting to what the user said - Premake needs to get out of the way.

Reply:
@samsinsane Agreed on all points, but this was just intended as a placeholder to get things working—I had a comment to indicate that in the VS version that didn't get carried over. I believe (but don't yet know for sure) that these will end up in the eventual toolset abstraction, and will be extensible from there. Once we have baseline exporters for VS (done), Gmake and Xcode (in progress) that is on my shortlist to dig into.